operator-common/src/main/java/io/strimzi/operator/common/operator/resource/AbstractResourceOperator.java

```diff
@@ -127,16 +130,34 @@ public AbstractResourceOperator(Vertx vertx, C client, String resourceKind) {
      * and completes the given future accordingly.
      */
     protected Future<ReconcileResult<T>> internalPatch(String namespace, String name, T current, T desired) {
+        return internalPatch(namespace, name, current, desired, true);
+    }
+
+    protected Future<ReconcileResult<T>> internalPatch(String namespace, String name, T current, T desired, boolean cascading) {
         try {
-            ReconcileResult.Patched<T> result = ReconcileResult.patched(operation().inNamespace(namespace).withName(name).cascading(true).patch(desired));
+            T result = operation().inNamespace(namespace).withName(name).cascading(cascading).patch(desired);
             log.debug("{} {} in namespace {} has been patched", resourceKind, name, namespace);
-            return Future.succeededFuture(result);
+            return Future.succeededFuture(wasChanged(current, result) ? ReconcileResult.patched(result) : ReconcileResult.noop(result));
         } catch (Exception e) {
             log.error("Caught exception while patching {} {} in namespace {}", resourceKind, name, namespace, e);
             return Future.failedFuture(e);
         }
     }
 
+    private boolean wasChanged(T oldVersion, T newVersion) {
+        if (oldVersion != null && newVersion != null) {
+            try {
+                Object x = oldVersion.getClass().getMethod("getSpec").invoke(oldVersion);
+                Object y = newVersion.getClass().getMethod("getSpec").invoke(newVersion);
+                return JsonDiff.asJson(patchMapper().valueToTree(x), patchMapper().valueToTree(y)).iterator().hasNext();
```

Comment:
isn't it possible that the `current` (so `oldVersion`) got from the status of the deployment differs from the `desired` (so `newVersion`) for fields update at runtime by K8S? Asking because I'm confused about the need for a `StatefulSetsDiff` with paths to ignore but just a JSON diff here.

Reply:
Thinking about it this is utter rubbish and there's a much better way.

Reply:
I've pushed a much neater way to do this, and it's super-obvious when you think about it: Just use the `metadata.resourceVersion` on the returned resource to determine whether the resource was changed. This relies on fabric8's diffing to basically send an empty patch if the resources are the same, in which case the `resourceVersion` won't be changed. This avoids having to compute a diff ourselves and should also avoid the problem of the k8s-maintained metadata being different.